components/list/list-item-checkbox-mixin.js

```diff
@@ -100,13 +100,26 @@ export const ListItemCheckboxMixin = superclass => class extends SkeletonMixin(L
 		if (!suppressEvent) this._dispatchSelected(selected);
 	}
 
-	_dispatchSelected(value) {
+	async _dispatchSelected(value) {
+		/* wait for internal state to be updated in case of action-click case so that a consumer
+		 calling getSelectionInfo will get the correct state */
+		await this.updateComplete;
 		this.dispatchEvent(new CustomEvent('d2l-list-item-selected', {
 			detail: { key: this.key, selected: value },
 			bubbles: true
 		}));
 	}
 
+	_getNestedList() {
```

Comment:
I originally factored this out so that it could be used to evaluate the selected keys for a tree or sub-tree, but in the end, I found a much more performant method.  No harm in this factoring though, so I kept this change.